### PR TITLE
COMP: Provide basic completion via completion contributor

### PR DESCRIPTION
**Pull request by mchernyavsky**
This is needed for another [PR](https://github.com/intellij-rust/intellij-rust/pull/3145).
It should also improve the completion response time, because now completion variants will appear in batches.

TODO:
* Add completion-based performance tests

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/completion/RsBasicCompletionProvider.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+import org.rust.lang.core.psiElement
+import org.rust.lang.core.resolve.*
+import org.rust.lang.core.resolve.ref.MethodResolveVariant
+import org.rust.lang.core.types.infer.containsTyOfClass
+import org.rust.lang.core.types.ty.Ty
+import org.rust.lang.core.types.ty.TyTypeParameter
+import org.rust.lang.core.types.ty.TyUnknown
+import org.rust.lang.core.types.type
+
+object RsBasicCompletionProvider : CompletionProvider<CompletionParameters>() {
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/completion/RsBasicCompletionProvider.kt]: Maybe "RsCommonCompletionProvider"

**Comment by vlad20012** [comment]
I think we can merge it as is and ring performance tests later

**Comment by mchernyavsky** [comment]
Done.

**Pull request merged**